runners/samza/src/main/java/org/apache/beam/runners/samza/translation/GroupByKeyTranslator.java

```diff
@@ -228,9 +225,63 @@ private static <K, InputT, OutputT> MessageStream<OpMessage<KV<K, OutputT>>> doT
                         reduceFn,
                         windowingStrategy,
                         new DoFnOp.SingleOutputManagerFactory<>(),
-                        transformFullName,
-                        transformId,
-                        isBounded)));
+                        ctx.getTransformFullName(),
+                        ctx.getTransformId(),
+                        isBounded),
+                    ctx));
+    return outputStream;
+  }
+
+  private static <K, InputT, OutputT> MessageStream<OpMessage<KV<K, OutputT>>> doTranslateGBK(
```

Comment:
Are we duplicating the entire doTranslateGBK(). why we are doing this?

Reply:
https://github.com/apache/beam/blob/0474afbe608cc6e43f8acc3464005e98c8ff4555/runners/samza/src/main/java/org/apache/beam/runners/samza/translation/PortableTranslationContext.java#L51 

For open source the PortableTranslationContext & TranslationContext do not have a lineage, this change was made to LinkedIn fork of beam but was not merged to open source